### PR TITLE
History tab creation

### PR DESCRIPTION
**Pull request by melnelen**

### CODE REVIEW

**Comment by melnelen** [comment]
https://user-images.githubusercontent.com/8265393/171202951-1cf72b1f-3518-4c9a-9d6f-34ab873f166c.mp4

File: FocusOn/Services/DataServiceProtocol.swift [hunk 1]
```diff
@@ -8,8 +8,9 @@
 import Foundation
 
 protocol DataServiceProtocol {
+    var allGoals: [Goal]? { get }
     func upsertGoals(goal: Goal, name: String, isCompleted: Bool)
-    func fetchGoals() -> [Goal]
+    func fetchGoals() -> [Goal]?
```
Comment (cyril-lepottier) [inline FocusOn/Services/DataServiceProtocol.swift]: Why the decision of optional instead of an empty array?

**Review by cyril-lepottier — APPROVED** [review]
It misses the tests on `HistoryViewModel`, appart from that, it seems it's going very well :)

File: FocusOn/Services/MockDataService.swift [hunk 2]
```diff
@@ -8,18 +8,41 @@
...
-    func fetchGoals() -> [Goal] {
-        return savedGoals
+    func fetchGoals() -> [Goal]? {
+        return allGoals
     }
 
     func insertGoal(goal: Goal) throws {
         try checkLength(of: goal.name)
-        savedGoals.append(goal)
+        if allGoals != nil {
```
Comment (cyril-lepottier) [inline FocusOn/Services/MockDataService.swift]: if let allGoals = allGoals syntax is cleaner

**Review by cyril-lepottier — APPROVED** [review]
All good

**Pull request merged**